File: Aula_Ricardo/Programação_orientada_a_objetos/python/Herança.py
class Animal:

    def som(self):
        raise NotImplementedError("")
    
    
class Cachorro(Animal):
    def som(self):
        return "auau"

class Gato(Animal):
    def som(self):
        return "miauu"


if __name__ == "__main__":
    cachorro = Cachorro()
    gato = Gato()
    print(f"Cachorro: {cachorro.som()}")
    print(f"Gato: {gato.som()}")
    print(gato.som())

File: Aula_Ricardo/Programação_orientada_a_objetos/java/Herança.java
abstract class Animal {
    public abstract String som();
}

class Cachorro extends Animal {
    @Override
    public String som() {
        return "auau";
    }
}

class Gato extends Animal {
    @Override
    public String som() {
        return "miauu";
    }
}

public class Main {
    public static void main(String[] args) {
        Animal cachorro = new Cachorro();
        Animal gato = new Gato();
        
        System.out.println("Cachorro: " + cachorro.som());
        System.out.println("Gato: " + gato.som());
    }
}
